apps/site/assets/ts/stop/__tests__/components/amenities/ParkingAmenityCardTest.tsx

```diff
@@ -1,10 +1,148 @@
+  });
+
+  it("should list each lots daily, and monthly cost", async () => {
+    const user = userEvent.setup();
+    const localTestStop = { ...testStop, parking_lots: testLots };
+    render(<ParkingAmenityCard stop={localTestStop} alertsForParking={[]} />);
+    await user.click(screen.getByRole("button"));
+    expect(screen.getByText("$5")).toBeInTheDocument();
+    expect(screen.getByText("$300")).toBeInTheDocument();
+
+    expect(screen.getByText("Free")).toBeInTheDocument();
+    expect(screen.getByText("$100")).toBeInTheDocument();
+  });
+
+  it("should list each parking lots overnight status", async () => {
+    const user = userEvent.setup();
+    const localTestStop = { ...testStop, parking_lots: testLots };
+    render(<ParkingAmenityCard stop={localTestStop} alertsForParking={[]} />);
+    await user.click(screen.getByRole("button"));
+    expect(screen.getByText("Available")).toBeInTheDocument();
+    expect(screen.getByText("Not available")).toBeInTheDocument();
+  });
+
+  it("should list each parking lots capacity", async () => {
+    const user = userEvent.setup();
+    const localTestStop = { ...testStop, parking_lots: testLots };
+    render(<ParkingAmenityCard stop={localTestStop} alertsForParking={[]} />);
+    await user.click(screen.getByRole("button"));
+    expect(screen.getByText("543 total parking spots")).toBeInTheDocument();
+    expect(screen.getByText("123 accessible spots")).toBeInTheDocument();
+
+    expect(screen.getByText("44 total parking spots")).toBeInTheDocument();
+    expect(screen.getByText("0 accessible spots")).toBeInTheDocument();
+  });
+
+  it("should only list payment methods if each lot supports it", async () => {
+    const user = userEvent.setup();
+    const localTestStop = { ...testStop, parking_lots: testLots };
+    render(<ParkingAmenityCard stop={localTestStop} alertsForParking={[]} />);
+    await user.click(screen.getByRole("button"));
+    const paymentMethods = screen.queryAllByText("Payment Methods");
+    const payByPhoneArray = screen.queryAllByText("PayByPhone");
+    const invoiceArray = screen.getAllByText(/Invoice/);
+    expect(paymentMethods.length).toBe(2);
+    expect(payByPhoneArray.length).toBe(1);
+    expect(invoiceArray.length).toBe(1);
+    expect(payByPhoneArray[0]).toBeInTheDocument();
+    expect(invoiceArray[0]).toBeInTheDocument();
+    expect(screen.getByText(/Location 1234/)).toBeInTheDocument();
+  });
+
+  it("should only show a location link if the parking lot has a latitude and longitude", async () => {
+    const user = userEvent.setup();
+    const localTestStop = { ...testStop, parking_lots: testLots };
+    render(<ParkingAmenityCard stop={localTestStop} alertsForParking={[]} />);
+    await user.click(screen.getByRole("button"));
+    const directionLinks = screen.getAllByText(/Get directions/);
+    expect(directionLinks.length).toBe(1);
+    expect(directionLinks[0]).toBeInTheDocument();
```

Comment:
nitpick: Since the length is already checked and directionLinks was built by querying the page this `toBeInTheDocument` seems repetitive